DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1930,12 +1941,35 @@ internal async Task OnInteractionCreateAsync(ulong? guildId, ulong channelId, Tr
             }
             else
             {
-                var ea = new InteractionCreateEventArgs
+                if (interaction.Data.Target.HasValue) // Context-Menu. //
                 {
-                    Interaction = interaction
-                };
+                    var targetId = interaction.Data.Target.Value;
+                    DiscordUser targetUser = null;
+                    DiscordMember targetMember = null;
+                    DiscordMessage targetMessage = null;
+
+                    interaction.Data.Resolved.Messages?.TryGetValue(targetId, out targetMessage);
+                    interaction.Data.Resolved.Members?.TryGetValue(targetId, out targetMember);
+                    interaction.Data.Resolved.Users?.TryGetValue(targetId, out targetUser);
 
-                await this._interactionCreated.InvokeAsync(this, ea).ConfigureAwait(false);
+                    var ctea = new ContextMenuInteractionCreateEventArgs
+                    {
+                        Interaction = interaction,
+                        TargetUser = targetMember ?? targetUser,
+                        TargetMessage = targetMessage,
+                        Type = targetUser == null ? ApplicationCommandType.MessageContextMenu : ApplicationCommandType.UserContextMenu,
```

Comment:
Wasn't this supposed to use some field discord sends that they didn't before

Reply:
Do they send it now?

Reply:
Dunno, doesn't really matter I suppose

Reply:
The type is sent now, but from what I can tell that's still an *OK* way of doing it. Not ideal, but you've merged this now so